app/src/test/kotlin/co/netguru/baby/monitor/client/feature/client/home/SendBabyNameUseCaseTest.kt

```diff
@@ -0,0 +1,53 @@
+package co.netguru.baby.monitor.client.feature.client.home
+
+import co.netguru.baby.monitor.RxSchedulersOverrideRule
+import co.netguru.baby.monitor.client.common.proto.Message
+import co.netguru.baby.monitor.client.data.DataRepository
+import co.netguru.baby.monitor.client.data.client.ChildDataEntity
+import co.netguru.baby.monitor.client.feature.communication.websocket.RxWebSocketClient
+import com.google.gson.Gson
+import com.nhaarman.mockito_kotlin.*
+import io.reactivex.Completable
+import io.reactivex.Maybe
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+class SendBabyNameUseCaseTest {
+
+    @Rule
+    @JvmField
+    val schedulersRule = RxSchedulersOverrideRule()
+
+    private val childDataEntity = ChildDataEntity("", name = "name")
+    private val dataRepository: DataRepository = mock {
+        on { getChildData() }.doReturn(Maybe.just(childDataEntity))
+    }
+    private val message = "message"
+    private val gson: Gson = mock {
+        on { toJson(any<Message>()) }.doReturn(message)
+    }
+    private val rxWebSocketClient: RxWebSocketClient = mock {
+        on { send(any()) }.doReturn(Completable.complete())
+    }
+    private lateinit var sendBabyNameUseCase: SendBabyNameUseCase
+
+    @Before
+    fun setUp() {
+        sendBabyNameUseCase = SendBabyNameUseCase(dataRepository, gson)
+    }
+
+    @Test
+    fun `should send baby name message`() {
+        sendBabyNameUseCase
+            .streamBabyName(rxWebSocketClient)
```

Comment:
`.test()` instead of `subscribe()` can be called and then you are able to assert some result

Reply:
Yeah, although this is a completable so there isn't much to assert here ^^.

Reply:
You can check if everything completed without errors or with specific one `.test().assertCompleted()` or `.test().assertError()`